src/components/common/Dialogs/EditColonyDetailsDialog/EditColonyDetailsDialogForm.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState } from 'react';
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { ColonyRole, Id } from '@colony/colony-js';
+import { useFormContext } from 'react-hook-form';
+
+import AvatarUploader from '~shared/AvatarUploader';
+import {
+  ActionDialogProps,
+  DialogControls,
+  DialogHeading,
+  DialogSection,
+} from '~shared/Dialog';
+import {
+  Annotations,
+  HookFormInput as Input,
+  InputStatus,
+} from '~shared/Fields';
+import ColonyAvatar from '~shared/ColonyAvatar';
+// import NotEnoughReputation from '~dashboard/NotEnoughReputation';
+import Avatar from '~shared/Avatar';
+import { useActionDialogStatus } from '~hooks';
+
+import {
+  CannotCreateMotionMessage,
+  NoPermissionMessage,
+  PermissionRequiredInfo,
+} from '../Messages';
+
+import styles from './EditColonyDetailsDialogForm.css';
+
+const displayName =
+  'common.EditColonyDetailsDialog.EditColonyDetailsDialogForm';
+
+const MSG = defineMessages({
+  title: {
+    id: `${displayName}.title`,
+    defaultMessage: 'Edit colony details',
+  },
+  name: {
+    id: `${displayName}.name`,
+    defaultMessage: 'Colony name',
+  },
+  logo: {
+    id: `${displayName}.logo`,
+    defaultMessage: 'Colony Logo (Optional)',
+  },
+  permittedFormat: {
+    id: `${displayName}.permittedFormat`,
+    defaultMessage: 'Permitted format: .png or .svg',
+  },
+  annotation: {
+    id: `${displayName}.annotation`,
+    defaultMessage: `Explain why you’re editing the colony's details (optional)`,
+  },
+  invalidAvatarFormat: {
+    id: `${displayName}.invalidAvatarFormat`,
+    defaultMessage: `Image you tried to upload is in an invalid format`,
+  },
+  cannotCreateMotion: {
+    id: `${displayName}.cannotCreateMotion`,
+    defaultMessage: `Cannot create motions using the Governance v{version} Extension. Please upgrade to a newer version (when available)`,
+  },
+});
+
+const requiredRoles = [ColonyRole.Root];
+
+const EditColonyDetailsDialogForm = ({
+  back,
+  colony,
+  colony: { colonyAddress, metadata },
+  enabledExtensionData,
+}: ActionDialogProps) => {
+  const { setValue, getValues } = useFormContext();
+  const { colonyAvatarImage, colonyDisplayName } = getValues();
+  const [showUploadedAvatar, setShowUploadedAvatar] = useState(false);
+  const [avatarFileError, setAvatarFileError] = useState(false);
+
+  const { userHasPermission, canCreateMotion, disabledInput, disabledSubmit } =
+    useActionDialogStatus(
+      colony,
+      requiredRoles,
+      [Id.RootDomain],
+      enabledExtensionData,
+    );
+
+  /*
+   * Note that these threee methods just read the file locally, they don't actually
+   * upload it anywere.
+   *
+   * The upload method returns the file as a base64 string, which, after we submit
+   * the form we will be uploading to IPFS
```

Comment:
Is the plan to still upload to IPFS or to the DB how its done for the user avatar?